cpp/src/parquet/encoding.cc

```diff
@@ -863,6 +941,18 @@ std::unique_ptr<Encoder> MakeEncoder(Type::type type_num, Encoding::type encodin
         DCHECK(false) << "Encoder not implemented";
         break;
     }
+  } else if (encoding == Encoding::BYTE_STREAM_SPLIT) {
+    switch (type_num) {
+      case Type::FLOAT:
+        return std::unique_ptr<Encoder>(
+            new ByteStreamSplitEncoder<FloatType>(descr, pool));
+      case Type::DOUBLE:
+        return std::unique_ptr<Encoder>(
+            new ByteStreamSplitEncoder<DoubleType>(descr, pool));
+      default:
+        DCHECK(false) << "Encoder not implemented";
```

Comment:
Better to raise an error than crash out, no?

Reply:
Done. Added a test for invalid types.
There might be also some benefit of using it for integers but I have not tested on any data set, and that can be changed in the future if found useful.